File: src/Day9.java
import static java.lang.System.out;

public class Day9 {
    public static void main(String... args) {
        String line = IO.ReadLine("FILENAME").get(0);
        StringBuilder builder = new StringBuilder(line);
        int garbagecount = 0;
        while (line.contains("!")) {
            int i = line.indexOf("!");
            builder.deleteCharAt(i);
            builder.deleteCharAt(i);
            line = builder.toString();
        }
        while (line.contains("<")) {
            int start = line.indexOf("<");
            int end = line.indexOf(">");
            if (start != -1 && end != -1) {
                garbagecount += line.substring(start + 1, end).length();
                line = builder.delete(start, end + 1).toString();
            }
        }
        int counter=0,score = 0;
        for (int i = 0; i < line.length(); ++i) {
            if (line.charAt(i) == '{') {
                counter++;
                score += counter;
            } else if (line.charAt(i) == '}')
                counter--;
        }
        out.println("Score " + score);
        out.println("Garbage " +garbagecount);


    }
}

File: src/python/day9.py
with open("day9.txt") as file:
    garbage, jump = False, False;
    score, garbagecounter, counter = 0, 0 , 0;
    for x in file.read():
        if jump:
            jump = False
            continue
        if x == "!" :
            jump = True
        elif x == "<" and not garbage:
            garbage = True;
        elif x == ">":
            garbage = False;
        elif x == "{" and not garbage:
            counter += 1
            score += counter
        elif x == "}" and not garbage:
            counter -= 1
        elif garbage :
            garbagecounter += 1
    print("Part one "+ str(score))
    print("Part two "+str(garbagecounter))